rewrite-maven/src/test/kotlin/org/openrewrite/maven/RemoveUnneededDependencyOverridesTest.kt

```diff
@@ -0,0 +1,887 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import kotlin.io.writeText
+import kotlin.text.trimIndent
+
+class RemoveUnneededDependencyOverridesTest : MavenRecipeTest {
+
+    @Test
+    fun matchesParentDM_remove() {
+        val parent = """
+            <?xml version="1.0" encoding="UTF-8"?>
+            <project xmlns="http://maven.apache.org/POM/4.0.0"
+                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+                <modelVersion>4.0.0</modelVersion>
+            
+                <groupId>org.example</groupId>
+                <artifactId>parent-pom-test</artifactId>
+                <packaging>pom</packaging>
+                <version>1.0-SNAPSHOT</version>
+                <modules>
+                    <module>child-module1</module>
+                </modules>
+            
+                <dependencyManagement>
+                    <dependencies>
+                        <dependency>
+                            <groupId>com.google.guava</groupId>
+                            <artifactId>guava</artifactId>
+                            <version>30.0-jre</version>
+                        </dependency>
+                    </dependencies>
+                </dependencyManagement>
+            </project>
+        """.trimIndent()
+
+        val child = """
+            <?xml version="1.0" encoding="UTF-8"?>
+            <project xmlns="http://maven.apache.org/POM/4.0.0"
+                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+                <parent>
+                    <artifactId>parent-pom-test</artifactId>
+                    <groupId>org.example</groupId>
+                    <version>1.0-SNAPSHOT</version>
+                </parent>
+                <modelVersion>4.0.0</modelVersion>
+            
+                <artifactId>child-module-1</artifactId>
+                <packaging>pom</packaging>
+            
+                <dependencies>
+                    <dependency>
+                        <groupId>com.google.guava</groupId>
+                        <artifactId>guava</artifactId>
+                        <version>30.0-jre</version>
+                    </dependency>
+                </dependencies>
+            
+            </project>
+        """.trimIndent()
+
+        assertUnchanged(
+            recipe = RemoveUnneededDependencyOverrides(),
```

Comment:
If on the class itself you have `override val recipe = RemoveUnneededDependencyOverrides()` then you can skip specifying it on these `assertUnchanged` and [assertChanged](https://github.com/openrewrite/rewrite/blob/master/rewrite-test/src/main/kotlin/org/openrewrite/RecipeTest.kt#L46) calls as they default to using whatever is returned by `RecipeTest.getRecipe()`